packages/material-ui/src/Modal/Modal.js

```diff
@@ -10,7 +10,7 @@ import useForkRef from '../utils/useForkRef';
 import useEventCallback from '../utils/useEventCallback';
 import zIndex from '../styles/zIndex';
 import ModalManager, { ariaHidden } from './ModalManager';
-import TrapFocus from './TrapFocus';
+import TrapFocus from '../Unstable_TrapFocus/TrapFocus';
```

Comment:
To rely on the top index

```suggestion
import TrapFocus from '../Unstable_TrapFocus';
```